finder-app/finder.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+# recommended by deekshith patil on office hours
+
+if [ $# -ne 2 ];
+then
+   echo "Arguments requested not found, exiting...."
+   exit 1
+fi
+
+# https://youtube.com/watch?v=mXh7rObKU4E - Exit status and logging references
+
+if [ ! -d $1 ];
+then
+   echo "This is not a directory"
+   exit 1
+fi
+
+x=$(find $1 -type f | wc -l)
+y=$(grep -r $2 $1 | wc -l)
```

Comment:
@srki3050 for readability and also to make the implementation less dependent on argument definitions you may want to place in variables rather than just use the arguments directly.